internal/pipeline/runtime/container.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runtime
+
+import (
+	"fmt"
+	"io"
+	"os/user"
+)
+
+// ContainerNetworkName is a type for network name used in container
+type containerNetworkName string
+
+const (
+	networkNameNone containerNetworkName = "none"
+	networkNameHost containerNetworkName = "host"
+)
+
+type containerFunctionPermission struct {
+	AllowNetwork  bool
+	AsCurrentUser bool
+}
+
+// ContainerRunner implements a KRMFn which run a containerized
+// KRM function
+type ContainerRunner struct {
+	// Image is the container image to run
+	Image string
+
+	Exec ExecRunner
+
+	containerFunctionPermission
```

Comment:
Any reasons for preferring embedding vs explicit field `perms` ? I do want to refresh my notes on this topic.